packages/table/src/Cell/HeaderCell/HeaderCell.tsx

```diff
@@ -0,0 +1,88 @@
+import React, { PropsWithChildren } from 'react';
+
+import { cx } from '@leafygreen-ui/emotion';
+
+import { useTableContext } from '../../TableContext/TableContext';
+import { LGRowData } from '../../useLeafyGreenTable';
+import {
+  alignmentStyles,
+  baseCellStyles,
+  cellContentContainerStyles,
+  getCellPadding,
+} from '../Cell.styles';
+
+import SortIcon from './SortIcon/SortIcon';
+import {
+  getHeaderCellWidthStyles,
+  headerCellContentStyles,
+} from './HeaderCell.styles';
+import { HeaderCellProps, SortState, SortStates } from './HeaderCell.types';
+
+const HeaderSortState: SortStates = {
+  false: SortState.Off,
+  asc: SortState.Asc,
+  desc: SortState.Desc,
+};
+
+/**
+ * Component to wrap `<th>` elements for use inside `<thead>` elements.
+ */
+const HeaderCell = <T extends LGRowData>({
+  children,
+  className,
+  cellIndex,
+  header,
+  ...rest
+}: PropsWithChildren<HeaderCellProps<T>>) => {
+  const { table } = useTableContext();
+
+  const isFirstCell = cellIndex === 0;
+  const isSelectable = !!table && !!table.hasSelectableRows;
+
+  let columnName, sortState, onSortIconClick;
+
+  if (header && header.column.getCanSort()) {
+    columnName = header.column.columnDef.header as string;
+    const headerSortDirection = header.column.getIsSorted().toString();
```

Comment:
clicking the sort icon three times in a row takes it back to unsorted -- i wouldn't expect this. Clicking should oscillate between sorting a->z and z->a

Reply:
This was a design decision that Alven and I discussed, but if it just went a->z to z->a, then there'd be no way to unsort.

Reply:
Yeah just to add onto Sean's comment, the inability to unsort was previously brought up as an issue and this was the simplest solution we landed on. Other design systems use this pattern as well, here's [Carbon as an example](https://react.carbondesignsystem.com/?path=/story/components-datatable-sorting--default)